src/ngraph/ngraph_nnvm_ops.cc

```diff
@@ -148,7 +151,7 @@ void compute_backward(const mxnet::OpContext &ctx, std::shared_ptr<Graph> graph,
 
   // backward op
   std::vector<mxnet::NDArray> adjoints;
-  for (size_t i = 0; i < graph->num_outputs_; ++i) {
+  for (size_t i = 0; i < graph->num_adjoints_; ++i) {
```

Comment:
Would it make sense to replace these four lines with the following?
``` c++
std::vector<mxnet::NDArray> adjoints(inputs.begin(), inputs.begin() + graph->num_adjoints_);
```

Or if you _really_ want to go crazy, take it a step further and make it `const`, but I realize this is an unsettled difference of opinion:
``` c++
const std::vector<mxnet::NDArray> adjoints(inputs.begin(), inputs.begin() + graph->num_adjoints_);
```

Reply:
Yeah, probably.